lib/rspec/matchers/built_in/has.rb

```diff
@@ -56,11 +70,19 @@ def predicate_accessible?
           def private_predicate?
             @actual.private_methods.include? predicate.to_s
           end
+
+          def methods_include?(method)
+            @actual.methods.include?(method.to_s)
+          end
           # :nocov:
         else
           def private_predicate?
             @actual.private_methods.include? predicate
           end
+
+          def methods_include?(method)
+            @actual.methods.include?(method)
+          end
```

Comment:
One version of this method calls `to_s` on the method name, because the list of methods returned used to be strings.

We _could_ do that check inline (or look for either `method` _or_ `method.to_s`), it seemed in-line with the prior decision about `private_predicate` to just conditionally define the `methods_include?` method to handle that case - are you sure you want me to collapse that to an inline check? It'd look like one of these, and would be less performant:

```
@actual.methods.include?(method) || @actual.methods.include?(method.to_s)
@actual.methods.include?(String === methods.first ? method.to_s : method)
```

Reply:
Oh my bad, this is fine as is then.